### PR TITLE
chore: Fix cart / cart-core e2es

### PR DESCRIPTION
**Pull request by bgambocjaviniar**
closes https://jira.tools.sap/browse/CXSPA-1587

### CODE REVIEW

File: projects/storefrontapp-e2e-cypress/cypress/integration/regression/cart/cart.core-e2e-spec.ts [hunk 1]
```diff
@@ -75,6 +77,9 @@ describe('Cart', () => {
             'spartacus⚿electronics-spa⚿cart',
             JSON.stringify(storage)
           );
+
+          cy.wait(2000);
```
Comment (WeizhengSap) [inline projects/storefrontapp-e2e-cypress/cypress/integration/regression/cart/cart.core-e2e-spec.ts:81]: why here need wait 2s?

File: projects/storefrontapp-e2e-cypress/cypress/integration/regression/cart/cart.e2e-spec.ts [hunk 2]
```diff
@@ -233,26 +236,31 @@ describe('Cart', () => {
...
         cart.checkProductInCart(cart.products[1]);
 
-        // cleanup
-        cart.registerCartRefreshRoute();
         cart.removeCartItem(cart.products[0]);
-        cy.wait('@refresh_cart');
+        cy.wait('@refresh_cart').its('response.statusCode').should('eq', 200);
+
+        // flakes due to button still being disabled.
+        cy.wait(3000);
```
Comment (WeizhengSap) [inline projects/storefrontapp-e2e-cypress/cypress/integration/regression/cart/cart.e2e-spec.ts:260]: can we wait button enabled, instead of wait for 3s?

Comment (bgambocjaviniar) [reply projects/storefrontapp-e2e-cypress/cypress/integration/regression/cart/cart.e2e-spec.ts:260]: it still failed even with that added 'guard'

Comment (bgambocjaviniar) [reply projects/storefrontapp-e2e-cypress/cypress/integration/regression/cart/cart.e2e-spec.ts:260]: Screen Shot 2022-10-26 at 1 45 29 PM

https://dashboard.cypress.io/projects/k3nmep/runs/34751/overview/3ab781db-9136-4b9e-8cf2-ce75c8acd91e

Comment (bgambocjaviniar) [reply projects/storefrontapp-e2e-cypress/cypress/integration/regression/cart/cart.e2e-spec.ts:260]: reason I added a static wait was to let those other async actions to 'stabilize' before it gets to the button.

File: projects/storefrontapp-e2e-cypress/cypress/integration/regression/cart/cart.core-e2e-spec.ts (hunk 1 above)
Comment (bgambocjaviniar) [reply projects/storefrontapp-e2e-cypress/cypress/integration/regression/cart/cart.core-e2e-spec.ts:81]: reason for that, it flakes sometimes and not having a not found cart

the reason I can say is when cypress does this action by changing local storage, it usually just visits the cart part right away (very fast).

With this added static guard after setting the local storage, we make it wait to hopefully stabilize before we visit the cart page

Comment (bgambocjaviniar) [reply projects/storefrontapp-e2e-cypress/cypress/integration/regression/cart/cart.core-e2e-spec.ts:81]: You can find it working now from the last 8 runs https://dashboard.cypress.io/projects/k3nmep/runs?branches=%5B%7B%22label%22%3A%22feature%2FCXSPA-1587%22%2C%22suggested%22%3Afalse%2C%22value%22%3A%22feature%2FCXSPA-1587%22%7D%5D&committers=%5B%5D&flaky=%5B%5D&page=1&status=%5B%5D&tags=%5B%5D&timeRange=%7B%22startDate%22%3A%221970-01-01%22%2C%22endDate%22%3A%222038-01-19%22%7D

But all the runs you see from the dashboard that is filtered to this branch shows fixes according to the commit

**Review by WeizhengSap — APPROVED** [review]

**Pull request merged**